internal/oidc/oidc.go

```diff
@@ -15,16 +15,19 @@ type WellKnownEndpoints struct {
 }
 
 // GetWellKnownEndpointsFromIssuerURL gets the well known endpoints for the passed in issuer url.
-func GetWellKnownEndpointsFromIssuerURL(ctx context.Context, issuerURL url.URL) (*WellKnownEndpoints, error) {
+func GetWellKnownEndpointsFromIssuerURL(
+	ctx context.Context,
+	httpClient *http.Client,
```

Comment:
We are now passing this from the provider to ensure we can override it successfully with one that might have a retry mechanism.